pkgs/development/python-modules/qtile-extras/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools-scm
+}:
+
+buildPythonPackage rec {
+  pname = "qtile-extras";
+  version = "0.22.1";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "elParaguayo";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-2dMpGLtwIp7+aoOgYav2SAjaKMiXHmmvsWTBEIMKEW4=";
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  doCheck = false; # do not run tests because it can't find libqtile anyway
```

Comment:
Did you try to fix this? We probably only need to add it or patch dlopen or similar

Reply:
I did, but I couldn't get it to work. I added the following instead of `doCheck = false;`:
```nix
checkInputs = [
    pulseaudio
    qtile
  ] ++ (with python3Packages; [
    keyring
    requests
    stravalib
  ]);
```
and got this error, which I didn't manage to fix:
```
Found duplicated packages in closure for dependency 'cffi': 
  cffi 1.15.1 (/nix/store/1312ks4fn7bpsw0ivzfqhvzbnqwm06d4-python3.10-cffi-1.15.1/lib/python3.10/site-packages)
  cffi 1.15.1 (/nix/store/h9cl90r5w8v067wldzj62wbbgv8mqyvc-qtile-0.22.1/lib/python3.10/site-packages)
Found duplicated packages in closure for dependency 'pycparser': 
  pycparser 2.21 (/nix/store/vqlzj8qplrphr2yy3v6cnysm7q7y3mwl-python3.10-pycparser-2.21/lib/python3.10/site-packages)
  pycparser 2.21 (/nix/store/h9cl90r5w8v067wldzj62wbbgv8mqyvc-qtile-0.22.1/lib/python3.10/site-packages)
Found duplicated packages in closure for dependency 'python-dateutil': 
  python-dateutil 2.8.2 (/nix/store/h9cl90r5w8v067wldzj62wbbgv8mqyvc-qtile-0.22.1/lib/python3.10/site-packages)
  python-dateutil 2.8.2 (/nix/store/idh7gnb2062vl6j3ng19inisjark3x0p-python3.10-python-dateutil-2.8.2/lib/python3.10/site-packages)
Found duplicated packages in closure for dependency 'six': 
  six 1.16.0 (/nix/store/h9cl90r5w8v067wldzj62wbbgv8mqyvc-qtile-0.22.1/lib/python3.10/site-packages)
  six 1.16.0 (/nix/store/q70iagd39h98mrfq8zkhbswgx649ky8m-python3.10-six-1.16.0/lib/python3.10/site-packages)

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
error: builder for '/nix/store/z57cv8mrac7lywlpkaab0g2dk16j9m4q-python3.10-qtile-extras-0.22.1.drv' failed with exit code 1;
```

Reply:
You are getting the errors most likely because you added another python interpreter to inputs. Try adding the packages directly into the input attr.

Reply:
Hi, I've changed it now in the way you proposed, but sadly there is no difference. :(
I pushed the change to this PR now, would you be willing to take a look?

Thank you for helping! Sometimes I think I understand how Nix does things, but other times I'm completely at a loss...

Reply:
A bit more clarity: the error only happens when I add both `qtile` and a Python package like `requests` to `checkInputs`. So apparently `qtile` is what adds the additional Python environment.

Reply:
Okay, fixed it! The reason for the above error was that `qtile` is actually a wrapped package, so I had to use `qtile.unwrapped` in my `checkInputs`. So tests are running now, although 20 (out of 117) of them are failing due to various reasons. I've disabled those for now, is that okay, or should I investigate further?

Reply:
Do those tests all fail for similar reasons or are they completely different? Normally we try to group them and then add comments to the groups why those tests failed.

Reply:
Thanks again for your feedback. I have investigated further (and called in help from the `qtile-extras` developer), but I can't find a way to make those tests pass. I've grouped the failing tests now with a comment that describes the reason or error that makes them fail. What do you think about it now?

Reply:
@SuperSandro2000 sorry for pinging you, but do you have any more ideas about this PR?